core/Pos/Util/Justify.hs

```diff
@@ -0,0 +1,51 @@
+module Pos.Util.Justify
+       ( leftAlign
+       ) where
+
+import           Universum          hiding ((<>))
+
+import qualified Data.List          as List
+import qualified Data.List.NonEmpty as NonEmpty
+import           Data.Semigroup
+import qualified Data.Text          as Text
+
+data Line = Line
+  { lineWidth :: Int
+  , lineWords :: NonEmpty Text
+  }
+
+initLine :: Text -> Line
+initLine w = Line (Text.length w) (w :| [])
+
+appendLine :: Line -> Line -> Line
+appendLine line1 line2 =
+  Line
+    (lineWidth line1 + 1 + lineWidth line2)
+    (lineWords line1 <>    lineWords line2)
+
+nonEmptyTails :: NonEmpty a -> NonEmpty [a]
+nonEmptyTails = NonEmpty.fromList . List.tail . List.tails . NonEmpty.toList
```

Comment:
What about https://www.stackage.org/haddock/lts-9.1/base-4.9.1.0/Data-List-NonEmpty.html#v:tails?

Reply:
Not the same, it'll have an extra element in front

Reply:
I think it would be good to write a comment about it, because otherwise other people may also have this question when they see this code.